// react-chessground/chessground.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _chessground = require('chessground');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Chessground = function (_React$Component) {
  _inherits(Chessground, _React$Component);

  function Chessground() {
    _classCallCheck(this, Chessground);

    return _possibleConstructorReturn(this, _React$Component.apply(this, arguments));
  }

  Chessground.prototype.buildConfigFromProps = function buildConfigFromProps(props) {
    var config = { events: {} };
    Object.keys(Chessground.propTypes).forEach(function (k) {
      var v = props[k];
      if (v) {
        var match = k.match(/^on([A-Z]\S*)/);
        if (match) {
          config.events[match[1].toLowerCase()] = v;
        } else {
          config[k] = v;
        }
      }
    });
    return config;
  };

  Chessground.prototype.componentDidMount = function componentDidMount() {
    this.cg = (0, _chessground.Chessground)(this.el, this.buildConfigFromProps(this.props));
  };

  Chessground.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    this.cg.set(this.buildConfigFromProps(nextProps));
  };

  Chessground.prototype.componentWillUnmount = function componentWillUnmount() {
    this.cg.destroy();
  };

  Chessground.prototype.render = function render() {
    var _this2 = this;

    var props = { style: _extends({}, this.props.style) };
    if (this.props.width) {
      props.style.width = this.props.width;
    }
    if (this.props.height) {
      props.style.height = this.props.height;
    }
    return _react2.default.createElement('div', _extends({ ref: function ref(el) {
        return _this2.el = el;
      } }, props));
  };

  return Chessground;
}(_react2.default.Component);

Chessground.propTypes = {
  width: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.number]),
  height: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.number]),
  fen: _propTypes2.default.string,
  orientation: _propTypes2.default.string,
  turnColor: _propTypes2.default.string,
  check: _propTypes2.default.string,
  lastMove: _propTypes2.default.array,
  selected: _propTypes2.default.string,
  coordinates: _propTypes2.default.bool,
  autoCastle: _propTypes2.default.bool,
  viewOnly: _propTypes2.default.bool,
  disableContextMenu: _propTypes2.default.bool,
  resizable: _propTypes2.default.bool,
  addPieceZIndex: _propTypes2.default.bool,
  hightlight: _propTypes2.default.object,
  animation: _propTypes2.default.object,
  movable: _propTypes2.default.object,
  premovable: _propTypes2.default.object,
  predroppable: _propTypes2.default.object,
  draggable: _propTypes2.default.object,
  selectable: _propTypes2.default.object,
  onChange: _propTypes2.default.func,
  onMove: _propTypes2.default.func,
  onDropNewPiece: _propTypes2.default.func,
  onSelect: _propTypes2.default.func,
  items: _propTypes2.default.object,
  drawable: _propTypes2.default.object
};
Chessground.defaultProps = {
  coordinates: true,
  resizable: true,
  hightlight: {
    lastMove: true,
    check: true
  }
};
exports.default = Chessground;
